### DISCUSSION TITLE
Regex question

### DISCUSSION THREAD

**Discussion by DutchPete** [VSCodium/vscodium#886] [Q&A]
I use Obsidian for note taking, but since the app does not have a mass Search and Replace, I use VSCodium for that. (Note: I use VSCodium mostly for CSS coding).

In my notes I have this kind of footnote:
`⁺^[yyyy-mm-dd, [short text](URL) by John Smith]`

I would like to find all occurrences of this and replace it with just `⁺^[[yyyy-mm-dd, short text](URL)]`,

i.e. move the 2<sup>nd</sup> `[` bracket (the one before "short") to the left next to the 1<sup>st</sup> one, and delete the text "by John Smith" and its leading gap.

What regex syntax can I use for this?

**Comment by eCyclops** [comment]
It may not be the most concise format, but you can use
`(...)(.+?, )\[(.+?\(*.+?\)) .+\]`

`$1[$2$3]`

**Comment by DutchPete** [comment]
Thank you.
